includes/class-wc-stripe-api.php

```diff
@@ -227,34 +227,33 @@ public static function request( $request, $api = 'charges', $method = 'POST', $w
 		);
 
 		$response_headers = wp_remote_retrieve_headers( $response );
-		// Log the stripe version in the response headers, if present.
-		if ( isset( $response_headers['stripe-version'] ) ) {
-			WC_Stripe_Logger::log( "{$api} response with stripe-version: " . $response_headers['stripe-version'] );
-		}
 
 		if ( is_wp_error( $response ) || empty( $response['body'] ) ) {
-			WC_Stripe_Logger::log(
-				'Error Response: ' . print_r( $response, true ) . PHP_EOL . PHP_EOL . 'Failed request: ' . print_r(
-					[
-						'api'             => $api,
-						'request'         => $request,
-						'idempotency_key' => $idempotency_key,
-					],
-					true
-				)
+			// Stripe redacts API keys in the response.
+			WC_Stripe_Logger::error(
+				"Stripe API error: {$method} {$api}",
+				[
+					'request'         => $request,
+					'idempotency_key' => $idempotency_key,
+					'response'        => $response,
+				]
 			);
 
 			throw new WC_Stripe_Exception( print_r( $response, true ), __( 'There was a problem connecting to the Stripe API endpoint.', 'woocommerce-gateway-stripe' ) );
 		}
 
+		$response_body = json_decode( $response['body'] );
+
+		WC_Stripe_Logger::debug( "Stripe API response: {$method} {$api}", [ 'response' => $response_body ] );
```

Comment:
Following on from @wjrosa's [comment above](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4486/files#r2220208873), should we not log the response headers here as well?
```suggestion
		WC_Stripe_Logger::debug(
			"Stripe API response: {$method} {$api}",
			[
				'response_body'    => $response_body,
				'response_headers' => $response_headers,
			]
		);
```

Reply:
The headers do not contain any helpful information (even for errors); that's why it was not logged before. But I don't have a strong opinion about that, except that it adds up to the log size 🤷🏼

Reply:
That's fair -- if they don't include useful data, then we should not log them, as they are effectively noise.